lib/file-source/src/file_server.rs

```diff
@@ -92,7 +92,7 @@ where
         let mut known_small_files = HashSet::new();
 
         let mut existing_files = Vec::new();
-        for path in self.paths_provider.paths().into_iter() {
+        for path in self.paths_provider.paths().unwrap().into_iter() {
```

Comment:
Since we are at it, we should try to propagate this. Regular `?` should do the trick.

Reply:
There was a bit more dancing involved than just the `?`, but agree. Definitively a good measure.